internal/brokers/examplebroker/com.ubuntu.auth.ExampleBroker.xml

```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!-- This file should be added in /usr/share/dbus-1/interfaces/ to configure the ExampleBroker -->
+
+<!DOCTYPE node PUBLIC
+"-//freedesktop//DTD D-BUS Object Introspection 1.0//EN"
+"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd">
+
+<node>
+  <interface name="com.ubuntu.auth.ExampleBroker">
+    <method name="GetAuthenticationModes">
+      <arg type="s" direction="in" name="requestID"/>
+      <arg type="s" direction="in" name="username"/>
+      <arg type="s" direction="in" name="lang"/>
+      <arg type="a{s}s" direction="in" name="supportedUILayouts"/>
+      <arg type="s" direction="out" name="sessionID"/>
+      <arg type="s" direction="out" name="encryptionKey"/>
+      <arg type="a{s}s" direction="out" name="authenticationModes"/>
+    </method>
+    <method name="SelectAuthenticationMode">
+        <arg type="s" direction="in" name="requestID"/>
```

Comment:
An other option here, instead of using a request ID would be to instead make this method to return a `/com/ubuntu/auth/ExampleBroker/AuthMode/$ID/` object and that then provides the `IsAuthorized()` and `CancelRequest` methods.

So basically so you separate the BrokerManager from the BrokerAuthenticationObject.

Reply:
Hey @3v1n0. thanks for the suggestion! This interface definition was before we decided on how to handle the canceling part of the authentication process. Now that #4 is merged, only the `sessionID` is enough for us to know which request should be canceled.
However, it's still a very nice suggestion and we will probably draft something around it to see how it would look. Thanks!